### PR TITLE
feat: add cheatcode support

### PR DESCRIPTION
**Pull request by nbaztec**
# What :computer: 
* Adds cheatcode support

# Why :hand:
* Required for foundry feature parity

# Evidence :camera:
tests pass

### CODE REVIEW

File: src/transactions.rs [hunk 1]
```diff
@@ -1,11 +1,11 @@
 use era_test_node::{
-    fork::ForkDetails,
+    fork::{ForkDetails, ForkStorage},
     node::{
         InMemoryNode, InMemoryNodeConfig, ShowCalls, ShowGasDetails, ShowStorageLogs, ShowVMDetails,
     },
     system_contracts,
 };
-use multivm::interface::VmExecutionResultAndLogs;
+use multivm::{vm_latest::VmExecutionResultAndLogs, vm_refunds_enhancement::TracerPointer};
```
Comment (Deniallugo) [inline src/transactions.rs:8]: please keep it consistent. 
It's from `VmExecutionResultAndLogs` are from interface

File: src/transactions.rs [hunk 2]
```diff
@@ -201,8 +199,18 @@ where
         l2_tx.common_data.signature = PackedEthSignature::default().serialize_packed().into();
     }
 
+    let tracers: Vec<
+        TracerPointer<
+            StorageView<ForkStorage<&RevmDatabaseForEra<DB>>>,
+            multivm::vm_refunds_enhancement::HistoryDisabled,
+        >,
+    > = vec![Box::new(CheatcodeTracer::new())];
```
Comment (Deniallugo) [inline src/transactions.rs:207]: You can use method `into_tracer_pointer` and remove all boxes and types.

**Review by Deniallugo — CHANGES_REQUESTED** [review]

File: src/transactions.rs [hunk 3]
```diff
@@ -201,8 +200,18 @@ where
         l2_tx.common_data.signature = PackedEthSignature::default().serialize_packed().into();
     }
 
+    let tracers: Vec<
+        TracerPointer<
+            StorageView<ForkStorage<&RevmDatabaseForEra<DB>>>,
+            multivm::vm_refunds_enhancement::HistoryDisabled,
+        >,
```
Comment (Deniallugo) [inline src/transactions.rs:207]: You can now remove type annotations.

File: src/cheatcodes.rs [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+        function etch(address who, bytes calldata code)
+        function setNonce(address account, uint64 nonce)
+        function roll(uint256 blockNumber)
+        function warp(uint256 timestamp)
+    ]"#
+);
+
+#[derive(Clone, Debug, Default)]
+// #[derive(Debug)]
+pub struct CheatcodeTracer {}
```
Comment (Deniallugo) [inline src/cheatcodes.rs:35]:
```suggestion
pub struct CheatcodeTracer;
```

**Review by Deniallugo — APPROVED** [review]
Good start! let's move further!

File: .github/workflows/tests.yaml [hunk 5]
```diff
@@ -58,4 +58,22 @@ jobs:
       run: |
         git clone https://github.com/matter-labs/zkfoundry-smoke-test
         cd zkfoundry-smoke-test
+        git fetch && git checkout nish-cargo-lock-fix
```
Comment (MexicanAce) [inline .github/workflows/tests.yaml:61]: (To be removed) Let's remove this `git checkout` before merge

File: .github/workflows/tests.yaml [hunk 6]
```diff
@@ -58,4 +58,22 @@ jobs:
...
+          submodules: recursive
+          ref: ${{ github.event.pull_request.head.sha }}
+      
+      - name: Install Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: nightly-2023-07-23
+
+      - name: Test Cheatcodes
+        run: cd tests && ls -l && ls -l lib && ls -l lib/forge-std &&  ./test.sh
```
Comment (MexicanAce) [inline .github/workflows/tests.yaml:79]: Let's clean this `run` command before merge 😅

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**